src/types.ts

```diff
@@ -46,15 +46,37 @@ export type ResourceRelationshipData = {
   id: string;
 };
 
-export type Meta = {
-  [key: string]: AttributeValue;
+export type Meta = Record<string, AttributeValue>;
+
+export type Link = string | {
+  href: string;
+  meta?: Meta;
+};
+
+export type Links = {
+  self?: Link;
+  related?: Link;
```

Comment:
I think we should lose it up a bit because, in theory, it could contain any property.
For example https://github.com/json-api/json-api/pull/1515/files#diff-a80e2578f8c89d0702e3677e4f40fd400711ffd6be44a637a5e0b61560d378feL779

Reply:
We could use a `Partial<Link>` perhaps to lose it up a bit.

Reply:
That will only make them bot optional, which they already are. I was thinking to add some generic so we can extend it from the outside. 

Or something like this
 ` [Key: string]: T;`